src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToPatchService.java

```diff
@@ -41,6 +43,8 @@ public AmendmentBody convertToPatch(Transaction transaction) {
         //TODO: throw EdifactValidationException if this produces empty list but only after at least 1 amendment story is completed
         return patchTransactionMappers.stream()
             .map(patchTransactionMapper -> patchTransactionMapper.map(transaction))
+            .filter(Objects::nonNull)
+            .distinct()
```

Comment:
I think this distinct actually does nothing as mappers don't implement `equals()` method

Reply:
It is for given names, if one of the given names are removed, all three should be. So if any is removed, the patch to remove all given (not given/0, give/1, given/2) is created. But this patch in current implementation could be created three times. To avoid it, I can merge these three transaction patch mappers into one, then the distinct could be remove. I did not spot this solution before :)

Reply:
No, there should one only for removal, needs to be distinct patches for replacement.

Reply:
distinct does sth, because we use annotation to generate hash code and equals methods

Reply:
We can assume that if the Forename is erased in EDIFACT then no other forenames will be included